index.js

```diff
@@ -191,78 +193,84 @@ module.exports = {
           "static-methods",
           "lifecycle",
           "everything-else",
-          "rendering"
+          "rendering",
         ],
-        groups: { rendering: ["/^render.+$/", "render"] }
-      }
+        groups: { rendering: ["/^render.+$/", "render"] },
+      },
     ],
-    "react/no-redundant-should-component-update": "error",
-    "react/no-will-update-set-state": "error",
-    "react/no-unsafe": "error",
+    "react/style-prop-object": "error",
     "react/void-dom-elements-no-children": "error",
 
-    // turn off prop types validation, better use ts ;)
-    "react/prop-types": "off",
-
-    // TODO turn all remaining rules to "error" eventually
+    // turn all remaining rules to "error" eventually
     "react/jsx-no-useless-fragment": "warn",
     "react/no-array-index-key": "warn",
     "react/no-danger": "warn",
+    "react/function-component-definition": [
+      "warn",
+      {
+        namedComponents: "function-declaration",
+        unnamedComponents: "function-expression",
+      },
+    ],
```

Comment:
This one might raise issues on SQ, if it's the case we can discuss if we want to keep it or not